### ISSUE TITLE
Werror=undef gives error _HAVE_STDINT_H not defined

### ISSUE THREAD

**Issue by arbruijn** [libsdl-org/SDL#6619]
Starting with SDL2 2.26.0, when including SDL.h with compiler option -Werror=undef the following compilation error is returned:
`SDL_config.h:548:58: error: "_HAVE_STDINT_H" is not defined, evaluates to 0`

It seems to be caused by this change in fd93f81
```diff
-#  if !defined(_STDINT_H_) && !defined(HAVE_STDINT_H) && !defined(_HAVE_STDINT_H)
+#if !defined(_STDINT_H_) && (!defined(HAVE_STDINT_H) || !_HAVE_STDINT_H)
````

**Comment by sezero** [comment]
CC: @ccawley2011

**Comment by ccawley2011** [comment]
Changing that line back should be OK, although I wonder why it didn't produce warnings with the standalone MSVC projects or the OpenWatcom makefile.

**Comment by sezero** [comment]
Please send a patch.

What I do wonder is who the hell defines `_HAVE_STDINT_H` with that leading prefix..

**Comment by slouken** [comment]
> What I do wonder is who the hell defines `_HAVE_STDINT_H` with that leading underscore..

Nobody. Removed. :)

**Comment by arbruijn** [comment]
Any chance the fix could be included in a 2.26.x release?

**Comment by slouken** [comment]
Yep, this will be in 2.26.2, fixed in 2d86b4ef7
